src/integration/java/io/pinecone/integration/controlPlane/pod/CreateDescribeListAndDeleteIndexTest.java

```diff
@@ -1,51 +1,181 @@
 
-        // Delete the index
-        controlPlaneClient.deleteIndex(indexName);
-        Thread.sleep(3500);
+    @Test
+    public void createIndexWithPodsAndPodType() {
+        String podIndexName = RandomStringBuilder.build("create-pod", 8);
+        CreateIndexRequestSpecPod podSpec = new CreateIndexRequestSpecPod().environment(environment).pods(2).podType(
+                "p1.x2");
+        CreateIndexRequestSpec createIndexRequestSpec = new CreateIndexRequestSpec().pod(podSpec);
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest()
+                .name(podIndexName)
+                .metric(IndexMetric.COSINE)
+                .dimension(10)
+                .spec(createIndexRequestSpec);
+
+        IndexModel createdIndex = controlPlaneClient.createIndex(createIndexRequest);
+        assertEquals(createdIndex.getName(), podIndexName);
+        assertEquals(createdIndex.getSpec().getPod().getPods(), 2);
+        assertEquals(createdIndex.getSpec().getPod().getPodType(), "p1.x2");
+        assertEquals(createdIndex.getStatus().getReady(), false);
+        assertEquals(createdIndex.getStatus().getState(), IndexModelStatus.StateEnum.INITIALIZING);
+
+        controlPlaneClient.deleteIndex(podIndexName);
+    }
+
+    @Test
+    public void createIndexWithInvalidName() {
+        CreateIndexRequestSpecPod podSpec = new CreateIndexRequestSpecPod().environment(environment).podType("p1.x1");
+        CreateIndexRequestSpec createIndexRequestSpec = new CreateIndexRequestSpec().pod(podSpec);
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest()
+                .name("Invalid-name")
+                .metric(IndexMetric.COSINE)
+                .dimension(10)
+                .spec(createIndexRequestSpec);
+
+        try {
+            controlPlaneClient.createIndex(createIndexRequest);
+
+            fail("Expected to throw PineconeBadRequestException");
+        } catch (PineconeBadRequestException expected) {
+            assertTrue(expected.getLocalizedMessage().contains("must consist of lower case alphanumeric characters or" +
+                    " '-'"));
+        }
+    }
+
+    @Test
+    public void createIndexWithInvalidDimension() {
+        CreateIndexRequestSpecPod podSpec = new CreateIndexRequestSpecPod().environment(environment).podType("p1.x1");
+        CreateIndexRequestSpec createIndexRequestSpec = new CreateIndexRequestSpec().pod(podSpec);
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest()
+                .name("invalid-dimension")
+                .metric(IndexMetric.COSINE)
+                .dimension(-1)
+                .spec(createIndexRequestSpec);
+
+        try {
+            controlPlaneClient.createIndex(createIndexRequest);
+
+            fail("Expected to throw PineconeUnmappedHttpException");
+        } catch (PineconeUnmappedHttpException expected) {
+            assertTrue(expected.getLocalizedMessage().contains("dimension: invalid value"));
```

Comment:
Maybe this error message could be more helpful if it said what is invalid about the user's dimension. So something like "Dimension invalid, must be integer > 0" or something